### PR TITLE
Use StatusCode instead of hardcoded integers

### LINKED ISSUE #10642: Documentation uses ProducesResponseType with numeric codes
I'm noticing that a lot of your documentation, uses magic numbers for the ProducesResponseType attribute.   Your Analyzer uses the StatusCodes enumeration, which I would assume, should be preferred.

[ProducesResponseType(404)]
[ProducesResponseType(StatusCodes.Status404NotFound)]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b391017f-25e1-d5e1-c4f7-020408ade7b5
* Version Independent ID: 263a7b64-2e42-bbfc-c1d3-61680438e1e4
* Content: [Build web APIs with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-2.2)
* Content Source: [aspnetcore/web-api/index.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/web-api/index.md)
* Product: **aspnet-core**
* GitHub Login: @scottaddie
* Microsoft Alias: **scaddie**

### PR DESCRIPTION
**Pull request by bartlannoeye**
Fixes #10642 

Continues on #10885 as I ran into that issue while getting to the StatusCode sample for my colleagues and this updates some of those added status codes as well.

### CODE REVIEW

File: aspnetcore/web-api/define-controller/samples/WebApiSample.Api.21/Controllers/PetsController.cs [hunk 1]
```diff
@@ -19,13 +20,15 @@ public PetsController(PetsRepository repository)
...
         [HttpGet]
+        [ProducesResponseType(typeof(IEnumerable<Pet>), StatusCodes.Status200OK)]
         public async Task<ActionResult<List<Pet>>> GetAllAsync()
         {
             return await _repository.GetPetsAsync();
         }
 
         [HttpGet("{id}")]
-        [ProducesResponseType(404)]
+        [ProducesResponseType(typeof(Pet), StatusCodes.Status200OK)]
```
Comment (scottaddie) [inline aspnetcore/web-api/define-controller/samples/WebApiSample.Api.21/Controllers/PetsController.cs]:
```suggestion
```

File: aspnetcore/web-api/define-controller/samples/WebApiSample.Api.21/Controllers/PetsController.cs [hunk 2]
```diff
@@ -19,13 +20,15 @@ public PetsController(PetsRepository repository)
         }
 
         [HttpGet]
+        [ProducesResponseType(typeof(IEnumerable<Pet>), StatusCodes.Status200OK)]
```
Comment (scottaddie) [inline aspnetcore/web-api/define-controller/samples/WebApiSample.Api.21/Controllers/PetsController.cs]:
```suggestion
```

File: aspnetcore/web-api/define-controller/samples/WebApiSample.Api.21/Controllers/ProductsController.cs [hunk 3]
```diff
@@ -38,6 +39,7 @@ public async Task<ActionResult<Product>> GetByIdAsync(int id)
 
         #region snippet_BindingSourceAttributes
         [HttpGet]
+        [ProducesResponseType(typeof(List<Product>), StatusCodes.Status200OK)]
```
Comment (scottaddie) [inline aspnetcore/web-api/define-controller/samples/WebApiSample.Api.21/Controllers/ProductsController.cs]:
```suggestion
```

File: aspnetcore/web-api/define-controller/samples/WebApiSample.Api.22/Controllers/ProductsController.cs [hunk 4]
```diff
@@ -17,8 +18,8 @@ public ProductsController(ProductsRepository repository)
         }
 
         [HttpGet("{id}")]
-        [ProducesResponseType(200)]
-        [ProducesResponseType(404)]
+        [ProducesResponseType(typeof(Product), StatusCodes.Status200OK)]
```
Comment (scottaddie) [inline aspnetcore/web-api/define-controller/samples/WebApiSample.Api.22/Controllers/ProductsController.cs]:
```suggestion
```

File: aspnetcore/web-api/define-controller/samples/WebApiSample.Api.22/Controllers/ProductsController.cs [hunk 5]
```diff
@@ -34,6 +35,7 @@ public async Task<ActionResult<Product>> GetByIdAsync(int id)
         }
 
         [HttpGet]
+        [ProducesResponseType(typeof(List<Product>), StatusCodes.Status200OK)]
```
Comment (scottaddie) [inline aspnetcore/web-api/define-controller/samples/WebApiSample.Api.22/Controllers/ProductsController.cs]:
```suggestion
```

File: aspnetcore/web-api/define-controller/samples/WebApiSample.Api.22/Controllers/ProductsController.cs [hunk 6]
```diff
@@ -52,8 +54,8 @@ public async Task<ActionResult<Product>> GetByIdAsync(int id)
         }
 
         [HttpPost]
-        [ProducesResponseType(201)]
-        [ProducesResponseType(400)]
+        [ProducesResponseType(typeof(Product), StatusCodes.Status201Created)]
```
Comment (scottaddie) [inline aspnetcore/web-api/define-controller/samples/WebApiSample.Api.22/Controllers/ProductsController.cs]:
```suggestion
        [ProducesResponseType(StatusCodes.Status201Created)]
```

File: aspnetcore/web-api/define-controller/samples/WebApiSample.Api.21/Controllers/ProductsController.cs [hunk 7]
```diff
@@ -21,8 +22,8 @@ public ProductsController(ProductsRepository repository)
 
         #region snippet_GetById
         [HttpGet("{id}")]
-        [ProducesResponseType(200)]
-        [ProducesResponseType(404)]
+        [ProducesResponseType(typeof(Product), StatusCodes.Status200OK)]
```
Comment (scottaddie) [inline aspnetcore/web-api/define-controller/samples/WebApiSample.Api.21/Controllers/ProductsController.cs]:
```suggestion
```

**Comment by bartlannoeye** [comment]
Rebased and applied comments

**Comment by scottaddie** [comment]
@bartlannoeye Thank you once again for all of your help!

**Pull request merged**